### ISSUE TITLE
Need help for NN training

### ISSUE THREAD

**Issue by imchyna** [torch/torch7#912]
Hi, all~

I wrote a nn modules to perform regression, the module is as below:

require 'nn'
require 'optim'
require 'torch'
src = torch.load('trainer.t7');
src_size = src.data:size(1);
perc = 0.0001;
train_size = math.floor(src_size*perc);
test_size = src_size-train_size;
trainset = {};
testset = {};
trainset.data = src.data[{{1,train_size}}];
trainset.label = src.label[{{1,train_size}}]
testset.data = src.data[{{train_size+1,src_size}}]
testset.label = src.label[{{train_size+1,src_size}}]

setmetatable (trainset,
      {_index = function(t,i)
            return t.data[i],t.label[i]
            end}
    )
trainset.data = trainset.data:double() 
trainset.label = torch.reshape(trainset.label,train_size):byte();
function trainset:size()
    return self.data:size()
end

net = nn.Sequential();
net:add(nn.SpatialConvolution(3,50,7,7))
net:add(nn.Identity())
net:add(nn.SpatialMaxPooling(26,26,26,26))--maxpooling
net:add(nn.Reshape(50*1))
net:add(nn.Linear(50*1,800))
net:add(nn.ReLU())
net:add(nn.Linear(800*1,800))
net:add(nn.Linear(800,1))
criterion = nn.MSECriterion( );
trainer = nn.StochasticGradient(net, criterion)
trainer.learningRate = 0.001
trainer.maxIteration = 5
trainer:train(trainset)

The procedure of "trainer:train(trainset)" has occured error, just like:
invalid arguments: LongStorage 
expected arguments: [*LongTensor*] [Generator] long
stack traceback:
	[C]: at 0x7ffbed7999f0
	[C]: in function 'randperm'
	...ta/torch/install/share/lua/5.1/nn/StochasticGradient.lua:19: in function 'train'
	[string "..."]:6: in main chunk
	[C]: in function 'xpcall'
	/home/grasta/torch/install/share/lua/5.1/itorch/main.lua:209: in function </home/grasta/torch/install/share/lua/5.1/itorch/main.lua:173>
	/home/grasta/torch/install/share/lua/5.1/lzmq/poller.lua:75: in function 'poll'
	/home/grasta/torch/install/share/lua/5.1/lzmq/impl/loop.lua:307: in function 'poll'
	/home/grasta/torch/install/share/lua/5.1/lzmq/impl/loop.lua:325: in function 'sleep_ex'
	/home/grasta/torch/install/share/lua/5.1/lzmq/impl/loop.lua:370: in function 'start'
	/home/grasta/torch/install/share/lua/5.1/itorch/main.lua:381: in main chunk
	[C]: in function 'require'
	(command line):1: in main chunk
	[C]: at 0x00406670

I'm a newbie to torch, excuse me if this is too dumb.Can anyone help me figure out this ?
Thanks, 
Daisy

**Comment by imchyna** [comment]
{
  data : DoubleTensor - size: 32x3x32x32
  size : function: 0x41869850
  label : ByteTensor - size: 32
}

this is the results of "print(trainset)"

**Comment by ksharsha** [comment]
It is expecting the dataset to be a tensor of type long where as it looks like you have it in double type. Change it to long should fix this error.

**Comment by imchyna** [comment]
I'll try it. thank you for your help.

2017-03-06 18:12 GMT+08:00 Sree Harsha Kalli <notifications@github.com>:
